airmozilla/main/views.py

```diff
@@ -1114,6 +1115,100 @@ def item_pubdate(self, event):
         return event.start_time
 
 
+def related_content(request, slug):
+    event = get_object_or_404(Event, slug=slug)
+
+    es = pyelasticsearch.ElasticSearch(settings.RELATED_CONTENT_URL)
+
+    fields = ['title', 'tags']
+    if list(event.channels.all()) != [
+            Channel.objects.get(slug=settings.DEFAULT_CHANNEL_SLUG)]:
+        fields.append('channel')
+
+    mlt_query = {
+        "more_like_this": {
+            "fields": fields,
+            "docs": [
+                {
+                    "_index": settings.ELASTICSEARCH_PREFIX
+                    + settings.ELASTICSEARCH_INDEX,
+                    "_type": "event",
+                    "_id": event.id
+                }],
+            "min_term_freq": 1,
+            "max_query_terms": 20,
+        }
+    }
+
+    if request.user.is_active:
+        if is_contributor(request.user):
+            query = {
+                "query": {
+                    "filtered": {
+                        "query": mlt_query,
+                        "filter": {
+                            "must_not": {
+                                "term": {
+                                    "privacy": Event.PRIVACY_COMPANY
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        else:
+            query = {
+                "query": mlt_query
+            }
+    else:
+        query = {
+            "query": {
+                "filtered": {
+                    "query": mlt_query,
+                    "filter": {
+                        "bool": {
+                            "must": {
+                                "term": {"privacy": Event.PRIVACY_PUBLIC}
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    query['from'] = 0
+    query['size'] = settings.RELATED_CONTENT_SIZE
+    ids = []
+    hits = es.search(query, index='events')['hits']
+
+    for doc in hits['hits']:
+        print "\t", repr(doc['_source']['title']), doc['_id'], doc['_score']
+        ids.append(int(doc['_id']))
+
+    events = Event.objects.scheduled_or_processing() \
+        .filter(id__in=ids) \
+        .exclude(status=Event.STATUS_REMOVED)
```

Comment:
Perhaps this is not necessary. After all, the `scheduled_or_processing()` should already do the necessary filtering on status.